liblangutil/CharStream.cpp

```diff
@@ -144,3 +144,31 @@ string CharStream::singleLineSnippet(string const& _sourceCode, SourceLocation c
 
 	return cut;
 }
+
+// @p _line and @p _column should be 1-based, because it's not an index but a line/column number.
+optional<int> CharStream::translateLineColumnToPosition(int _line, int _column) const
+{
+	return translateLineColumnToPosition(m_source, _line, _column);
+}
+
+optional<int> CharStream::translateLineColumnToPosition(std::string const& _text, int _line, int _column)
+{
+	size_t offset = 0;
+
+	while (_line > 0)
+	{
+		if (auto const found = _text.find('\n', offset); found != _text.npos)
```

Comment:
Why not keep the assignment out of the `if` for better readability?
Also, found sounds like a boolean variable but it isn't one.
```suggestion
		size_t const nextNewlinePosition = _text.find('\n', offset);
		if (nextNewlinePosition != _text.npos)
```

Reply:
> Why not keep the assignment out of the if for better readability?

Ugh, "readability" is sometimes a little subjective. I can move it out though. `found` identifier wasn't the most intuitive choice. `foundIndex` would probably be most precise - not sure.

Reply:
For me what makes this hard to read is putting two instructions inside the `if` condition. I often miss the condition behind the variable declaration. I think that this pattern is only readable if the condition is just checking if the variable's value is true/false. And even then I'm not a big fan of assignments inside `if`s.